.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,175 @@
+      - name: Collect supported dialects
+        id: dialects-matrix
+        run: |
+          printf 'dialects=%s\n' "$(jq -c '[.[].shortName]' data/dialects.json)" >> $GITHUB_OUTPUT
+
+  benchmark_files:
+    needs: dialects
+    runs-on: ubuntu-latest
+    outputs:
+      dialect_benchmarks: ${{ steps.benchmarks.outputs.dialect_benchmarks }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Bowtie
+        uses: ./
+
+      - name: Collect Benchmark Files
+        id: benchmarks
+        run: |
+          results=()
+          dialects='${{ needs.dialects.outputs.dialects }}'
+          dialects=$(echo $dialects | jq -r '.[]')
+          results=()
+
+          for dialect in $dialects; do
+            output=$(bowtie filter-benchmarks -D "$dialect")
+
+            if [ -n "$output" ]; then
+              while IFS= read -r line; do
+                json_result=$(jq -nc --arg p "$dialect" --arg o "$line" '{ dialect: $p, benchmark: $o }')
+                results+=("$json_result")
+              done <<< "$output"
+            fi
+          done
+          final_json="$(jq -sc '.' <<< "${results[@]}")"
+          final_json=$(echo "$final_json" | jq -c '{ "include": . }')
+          echo $final_json >> $GITHUB_STEP_SUMMARY
+          echo "dialect_benchmarks=$final_json" >> $GITHUB_OUTPUT
+
+  run_benchmarks:
+    needs: benchmark_files
+    runs-on: ubuntu-latest
+    timeout-minutes: 720
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.benchmark_files.outputs.dialect_benchmarks) }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Bowtie
+        uses: ./
+
+      - name: Install pyperf dependency
+        run: |
+          python -m pip install pyperf
+
+      - name: Generate Benchmark Report
+        run: |
+          bowtie perf $(bowtie filter-implementations | sed 's/^/-i /') -b ${{ matrix.benchmark }} -D ${{ matrix.dialect }} -q > ${{ matrix.benchmark }}-file.json
+
+      - name: Store Unique Id for distinction
+        run: echo "unique_id=$(uuidgen)" >> $GITHUB_ENV
+
+      - name: Upload Benchmark file as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: benchmark-file-${{ matrix.dialect }}-${{ env.unique_id }}
+          path: ${{ matrix.benchmark }}-file.json
+
+  merge_benchmarks_into_single_report:
+    needs:
+      - dialects
+      - run_benchmarks
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        dialect: ${{ fromJson(needs.dialects.outputs.dialects) }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Download Benchmark Reports for a dialect
+        uses: actions/download-artifact@v4
+        with:
+          pattern: benchmark-file-${{ matrix.dialect }}-*
+          path: benchmarks-${{ matrix.dialect }}/
+          merge-multiple: true
+
+      - name: Merge Benchmark Reports for a dialect
+        run: |
+          python - <<EOF
+          import os
```

Comment:
This *definitely* seems like it should live in Bowtie and use Bowtie's benchmark types rather than some ad hoc code here.

Reply:
I had thought about it but figured that maybe having a command like `bowtie merge-benchmarks` can do this but would be have any usecase for it other than this or not was the case. I decided since we arent probably gonna have such a usecase. It was better to do it here itself for the time being. 

What do you feel ? @Julian

Reply:
It's possible -- but then it still would be better as some code living in Bowtie which used Bowtie's types (and had tests) than just ad hoc Python code which has no tests and no assurance it'll work if we change the output format ever.

Reply:
That seems reasonably OK, but if `combine_benchmark_reports` is already going to take paths (and not reports) then it seems reasonable that it returns a str and not a `BenchmarkReport` (and so it should just json.dumps + serializable its return value probably). Making it take an iterable and not a list also seems easily doable to avoid needing the `list` call.